pyschism/forcing/hycom/hycom2schism.py

```diff
@@ -37,27 +41,27 @@ def convert_longitude(ds):
     ds = ds.rename({'_lon_adjusted': lon_name})
     return ds
 
-def get_database(date, Bbox=None):
+@lru_cache
+def get_database(date):
     if date >= datetime(2018, 12, 4):
-        database = f'GLBy0.08/expt_93.0'
+        return f'GLBy0.08/expt_93.0'
     elif date >= datetime(2018, 1, 1) and date < datetime(2018, 12, 4):
-        database = f'GLBv0.08/expt_93.0'
+        return f'GLBv0.08/expt_93.0'
     elif date >= datetime(2017, 10, 1) and date < datetime(2018, 1, 1):
-        database = f'GLBv0.08/expt_92.9'
+        return f'GLBv0.08/expt_92.9'
     elif date >= datetime(2017, 6, 1) and date < datetime(2017, 10, 1):
-        database = f'GLBv0.08/expt_57.7'
+        return f'GLBv0.08/expt_57.7'
     elif date >= datetime(2017, 2, 1) and date < datetime(2017, 6, 1):
-        database = f'GLBv0.08/expt_92.8'
+        return f'GLBv0.08/expt_92.8'
     elif date >= datetime(2016, 5, 1) and date < datetime(2017, 2, 1):
-        database = f'GLBv0.08/expt_57.2'
+        return f'GLBv0.08/expt_57.2'
     elif date >= datetime(2016, 1, 1) and date < datetime(2016, 5, 1):
-        database = f'GLBv0.08/expt_56.3'
+        return f'GLBv0.08/expt_56.3'
     elif date >= datetime(1994, 1, 1) and date < datetime(2016, 1, 1):
-        database = f'GLBv0.08/expt_53.X/data/{date.year}'
-    else:
-        logger.info('No data for {date}')
-    return database
+        return f'GLBv0.08/expt_53.X/data/{date.year}'
+    raise ValueError(f'No GOFS data for {date} (input must be >= than 1994-01-01).')
 
+@lru_cache
 def get_idxs(date, database, bbox):
 
     if date.strftime("%Y-%m-%d") >= datetime.now().strftime("%Y-%m-%d"):
```

Comment:
Is it really programmatically safe to compare dates as strings? Comparison operators are implemented on datetime objects...

Reply:
Also, consider using `datetime.utcnow()` since `datetime.now()` actually uses the system's locale.

Reply:
Resolved in the latest version. Thanks.